docs/proposals/web_uploads.md

```diff
@@ -0,0 +1,34 @@
+# Add upload scheduler to webservice
+
+This is a proposal to facet `bundle upload` via the Datamon webservice (i.e. the `web` command).
+
+Additionally, so that the web service -- currently a frontend for `repo list`, `bundle list`, and `bundle list files`, all operations that typically complete in per-HTTP-request goroutines in about the expected timeframe for page loads in a 1.0-ish webservice -- can continue to be useful for a possible deploy to Kubernetes, faceting `bundle upload` necessarily includes implementing some at least rudimentary scheduling functionality:  The webservice will need to keep track of several ongoing uploads, each triggered by an HTTP-request, but with fire-and-forget (i.e. event) semantics rather than those of request-response (i.e. RPC).
+
+### data sources
+
+This proposal calls for an additional, optional flag to the `web` command, `--upload-source`, say, which can be present in arbitrary arities (to allow arbitrary numbers of upload sources -- is this possible with pflags?).  When this flag isn't present, the webservice will continue to function as before.  When a non-empty list of directories is passed in via this flag, an additional page will be available in the web ui that, at first, provides a simple file browser rooted at each of these directories.
+
+From within any navigable directory of the filebrowser-as-a-webservice, it will additionally be possible to click an "upload" button to schedule a `bundle upload` of the tree rooted at that directory.  The "upload" button will be attached to a form faceting a few additional `bundle upload` parameters:
```

Comment:
I think a similar result, with more powerful generalization, could be achieved by upgrading our RW mount,
making it capable of _incremental uploads_ (newly available capability with diamond workflow).

The browser use case would be covered by simply adding a "commit" button and possibly some nice to have visual hint to figure out new/changed files on a view of the local file system (i.e. URL like file://home/my-dataset).

There would be no "upload" button: just use native browser or possibly some pretty standard file-dropping JS widget, but no interaction with `web serve`. There would be no async logic or scheduling whatsoever.

Reply:
Proposed sample command: `datamon bundle mount mutable --incremental`: starts a a fuse mount with a diamond, which ID is dumped in some local dot file and may be reused (and renewed) by subsequent commit signals.

Your idea of the local web service could come in handy here to signal the commit by a HTTP command (e.g. POST http://localhost:1234/commit) rather than by an OS signal.

Reply:
`datamon bundle mount mutable --incremental` i agree:  incremental upload functionality would better be faceted behind a ui other than diamond uploads.

if the faceted incremental upload can be used a starting point for the sort web service described here, so much the better.  i'm viewing this comment as supplementary information about how the implementation details here could differ, not as a total replacement for the webservice being described, which is not envisioned a local service on users' laptops but rather a service that we could run in any or all of the various stages' clusters to provide something more like the Jupyter "cleanroom" discussed in standup.